src/EventLoop/Internal/AbstractDriver.php

```diff
@@ -285,12 +285,21 @@ public function getSuspension(): Suspension
         \assert($fiber !== $this->fiber);
 
         // Use current object in case of {main}
-        return $this->suspensions[$fiber ?? $this] ??= new DriverSuspension(
+        if (isset($this->suspensions[$fiber ?? $this])) {
+            if ($suspension = $this->suspensions[$fiber ?? $this]->get()) {
+                return $suspension;
+            }
+        }
```

Comment:
Couldn't this be simplified down to something like the following?

```suggestion
            if ($suspension = $this->suspensions[$fiber ?? $this]?->get()) {
                return $suspension;
            }
```

Reply:
accessing a non-existent key will throw an error, so must be check it

Reply:
`?->get()`

Reply:
just like this [3v4l.org](https://3v4l.org/29425#v8.2.3)

Reply:
[?->](https://php.watch/versions/8.0/null-safe-operator)

Reply:
`Fatal error: Uncaught Error: Object stdClass#2 not contained in WeakMap`

i think  the document has some wrong, [WeakMap::offsetGet
](https://www.php.net/manual/zh/weakmap.offsetget.php)